velox/exec/MergeJoin.cpp

```diff
@@ -89,10 +91,14 @@ void MergeJoin::initialize() {
     initializeFilter(joinNode_->filter(), leftType, rightType);
 
     if (joinNode_->isLeftJoin() || joinNode_->isAntiJoin() ||
-        joinNode_->isRightJoin() || joinNode_->isFullJoin()) {
+        joinNode_->isRightJoin() || joinNode_->isFullJoin() ||
+        joinNode_->isLeftSemiFilterJoin() ||
+        joinNode_->isRightSemiFilterJoin()) {
       joinTracker_ = JoinTracker(outputBatchSize_, pool());
     }
-  } else if (joinNode_->isAntiJoin()) {
+  } else if (
+      joinNode_->isAntiJoin() || joinNode_->isLeftSemiFilterJoin() ||
+      joinNode_->isRightSemiFilterJoin() || joinNode_->isFullJoin()) {
```

Comment:
seems duplicated for `isLeftSemiFilterJoin`,`isRightSemiFilterJoin`,`isFullJoin`